test/bats/e2e-provider.bats

```diff
@@ -58,8 +68,41 @@ export LABEL_VALUE=${LABEL_VALUE:-"test"}
   assert_success
 }
 
-@test "deploy e2e-provider secretproviderclass crd" {
-  envsubst < $BATS_TESTS_DIR/e2e_provider_v1alpha1_secretproviderclass.yaml | kubectl apply -f -
+@test "[v1alpha1] deploy e2e-provider secretproviderclass crd" {
+  kubectl create namespace test-v1alpha1 --dry-run=client -o yaml | kubectl apply -f -
+
+  envsubst <  $BATS_TESTS_DIR/e2e_provider_v1alpha1_secretproviderclass.yaml | kubectl apply -n test-v1alpha1 -f -
+
+  kubectl wait --for condition=established -n test-v1alpha1 --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/e2e-provider -n test-v1alpha1 -o yaml | grep e2e-provider"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+}
+
+@test "[v1alpha1] CSI inline volume test with pod portability" {
+  envsubst < $BATS_TESTS_DIR/pod-secrets-store-inline-volume-crd.yaml | kubectl apply -n test-v1alpha1 -f -
+  
+  kubectl wait --for=condition=Ready -n test-v1alpha1 --timeout=180s pod/secrets-store-inline-crd
+
+  run kubectl get pod/secrets-store-inline-crd -n test-v1alpha1
+  assert_success
+}
+
+@test "[v1alpha1] CSI inline volume test with pod portability - read secret from pod" {
+  wait_for_process $WAIT_TIME $SLEEP_TIME "kubectl exec secrets-store-inline-crd -n test-v1alpha1 -- cat /mnt/secrets-store/$SECRET_NAME | grep '${SECRET_VALUE}'"
+
+  result=$(kubectl exec secrets-store-inline-crd -n test-v1alpha1 -- cat /mnt/secrets-store/$SECRET_NAME)
+  [[ "${result//$'\r'}" == "${SECRET_VALUE}" ]]
+}
+
+@test "[v1alpha1] CSI inline volume test with pod portability - read key from pod" {
+  result=$(kubectl exec secrets-store-inline-crd -n test-v1alpha1 -- cat /mnt/secrets-store/$KEY_NAME)
+  result_base64_encoded=$(echo "${result//$'\r'}" | base64 ${BASE64_FLAGS})
+  [[ "${result_base64_encoded}" == *"${KEY_VALUE_CONTAINS}"* ]]
+}
+
+@test "deploy e2e-provider v1 secretproviderclass crd" {
```

Comment:
nit: can we prefix `v1` like we are doing for `[v1alpha1]`. I like the idea of adding version to test name. It's super helpful and readable while going through test prow runs.
eg.
```bash
ok 4 [v1alpha1] deploy e2e-provider secretproviderclass crd in 834ms
ok 5 [v1alpha1] CSI inline volume test with pod portability in 1772ms
ok 6 [v1alpha1] CSI inline volume test with pod portability - read secret from pod in 346ms
ok 7 [v1alpha1] CSI inline volume test with pod portability - read key from pod in 164ms
ok 8 [v1] deploy e2e-provider secretproviderclass crd in 553ms
```

Can we do this for remaining test as well? To denote test is running for which version?